src/coreclr/vm/dynamicmethod.cpp

```diff
@@ -997,15 +992,71 @@ void LCGMethodResolver::RecycleIndCells()
     }
 }
 
-void LCGMethodResolver::Destroy()
+bool LCGMethodResolver::TryDestroyCodeHeapMemory()
 {
-    CONTRACTL {
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+        PRECONDITION(FinalizerThread::IsCurrentThreadFinalizer());
+    }
+    CONTRACTL_END;
+
+    while (m_DynamicCodePointers != NULL)
+    {
+        // Remove the unwind information (if applicable)
+        void* recordCodePointer = m_DynamicCodePointers->m_pEntry;
```

Comment:
This can be null if the actual memory allocation failed. Do we need to skip null entries here?